hpx/lcos/dataflow.hpp

```diff
@@ -133,8 +133,10 @@ namespace hpx { namespace lcos { namespace detail
         void execute(std::false_type, Futures&& futures)
         {
             try {
+                Func func = std::move(func_);
```

Comment:
Is this temporary really necessary?

Reply:
We do the same in the future shared state. Just wanted to mirror what's there.